tests/test_runner/tests.py

```diff
@@ -734,3 +736,16 @@ def test_run_checks_passes_and_teardown_raises(self):
                     runner.run_tests(['test_runner_apps.sample.tests_sample.TestDjangoTestCase'])
             self.assertTrue(teardown_databases.called)
             self.assertFalse(teardown_test_environment.called)
+
+
+class DiscoverRunnerDeprecationTests(SimpleTestCase):
+    def test_extra_tests_deprecated(self):
+        runner = NoOpTestRunner(verbosity=0, interactive=False)
+        msg = 'extra_tests is deprecated and will be removed in Django 5.0.'
+        with self.assertWarnsMessage(RemovedInDjango50Warning, msg):
+            runner.build_suite(extra_tests=[])
```

Comment:
I would make the test of `build_suite()` and `run_tests()` separate test methods.